pyretailscience/plots/line.py

```diff
@@ -153,3 +242,134 @@ def plot(
         gu.add_source_text(ax=ax, source_text=source_text)
 
     return gu.standard_tick_styles(ax)
+
+
+def plot(
+    df: pd.DataFrame | pd.Series,
+    value_col: str | list[str] | None = None,
+    x_label: str | None = None,
+    y_label: str | None = None,
+    title: str | None = None,
+    x_col: str | None = None,
+    group_col: str | None = None,
+    ax: Axes | None = None,
+    source_text: str | None = None,
+    legend_title: str | None = None,
+    move_legend_outside: bool = False,
+    fill_na_value: float | None = None,
+    highlight: str | list[str] | None = None,
+    **kwargs: dict[str, any],
+) -> SubplotBase:
+    """Plots the `value_col` over the specified `x_col` or index, creating a separate line for each unique value in `group_col`.
+
+    This function supports both pandas DataFrames and Series as input. When a Series is provided,
+    the Series values are plotted against its index, and `value_col` must be None.
+
+    Args:
+        df (pd.DataFrame | pd.Series): The dataframe or series to plot. When a Series is provided,
+            it represents the values to plot against its index.
+        value_col (str, list of str, or None): The column(s) to plot. Must be None when df is a Series.
+            Required when df is a DataFrame.
+        x_label (str, optional): The x-axis label.
+        y_label (str, optional): The y-axis label.
+        title (str, optional): The title of the plot.
+        x_col (str, optional): The column to be used as the x-axis. If None, the index is used.
+        group_col (str, optional): The column used to define different lines.
+        legend_title (str, optional): The title of the legend.
+        ax (Axes, optional): Matplotlib axes object to plot on.
+        source_text (str, optional): The source text to add to the plot.
+        move_legend_outside (bool, optional): Move the legend outside the plot.
+        fill_na_value (float, optional): Value to fill NaNs with after pivoting.
+        highlight (str or list of str, optional): Line(s) to highlight. When using
```

Comment:
This is a different style than we normally use
```suggestion
        highlight (str | list[str], optional): Line(s) to highlight. When using
```